src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -47,11 +46,12 @@ public class Transaction : IEquatable<Transaction>, IInventory, IInteroperable
         public TransactionAttribute[] Attributes
         {
             get => attributes;
-            set { attributes = value; _cosigners = null; _hash = null; _size = 0; }
+            set { attributes = value; _signers = null; _hash = null; _size = 0; }
         }
 
-        private Dictionary<UInt160, Cosigner> _cosigners;
-        public IReadOnlyDictionary<UInt160, Cosigner> Cosigners => _cosigners ??= attributes.OfType<Cosigner>().ToDictionary(p => p.Account);
+        private Dictionary<UInt160, Signer> _signers;
+        public IReadOnlyDictionary<UInt160, Signer> Signers => _signers ??= attributes.OfType<Signer>().ToDictionary(p => p.Account);
```

Comment:
If we're to keep them in an array (so they'll be ordered), we can remove `OrderBy` in `GetScriptHashesForVerifying` also.